src/server/scripts/Outland/BlackTemple/black_temple.cpp

```diff
@@ -178,7 +178,22 @@ class npc_wrathbone_flayer : public CreatureScript
     }
 };
 
+class go_den_of_mortal_door : public GameObjectScript
+{
+public:
+    go_den_of_mortal_door() : GameObjectScript("go_den_of_mortal_door") { }
+
+    void OnGameObjectStateChanged(GameObject* go, uint32 state) override
+    {
+        if (state == GO_STATE_ACTIVE)
+            if(InstanceScript* instance = go->GetInstanceScript())
+                if (Creature* trigger = instance->GetCreature(DATA_BLACK_TEMPLE_TRIGGER))
+                    trigger->AI()->Talk(EMOTE_DEN_OF_MORTAL_DOOR_OPEN);
```

Comment:
Both approach has issues.

With SetBossState:

If server crash when you raid running Black Temple, when server back this door will be locked and "has no way" to open again (because 4 boss already dead)

Actual approach i think less harmful for players, they can continue running Black Temple, only will see message 2 times (if server crash/soft reset)...